spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-bus-config-sample/spring-cloud-gcp-pubsub-bus-config-sample-test/src/test/java/com/example/LocalSampleAppIntegrationTest.java

```diff
@@ -0,0 +1,131 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.concurrent.TimeUnit;
+
+import org.awaitility.Awaitility;
+import org.bouncycastle.util.io.TeeOutputStream;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.web.client.RestTemplate;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assume.assumeThat;
+
+/**
+ * Integration test for the config client with local config server.
+ *
+ * <p>Because the test brings in config server/config client source files into a single project,
+ * config server bootstrap need to be suppressed in the config client app instance.
+ *
+ * @author Elena Felder
+ *
+ * @since 1.2
+ */
+public class LocalSampleAppIntegrationTest {
+
+	/** Test value of {example.message} property. */
+	public static final String VALUE = "INTEGRATION TEST";
+
+	private RestTemplate restTemplate = new RestTemplate();
+
+	private static PrintStream systemOut;
+
+	private static ByteArrayOutputStream baos;
+
+	@BeforeClass
+	public static void prepare() throws Exception {
+		assumeThat(
+			"PUB/SUB integration tests are disabled. Use '-Dit.pubsub=true' to enable.",
+			System.getProperty("it.pubsub"), is("true"));
+
+		systemOut = System.out;
+		baos = new ByteArrayOutputStream();
+		TeeOutputStream out = new TeeOutputStream(systemOut, baos);
+		System.setOut(new PrintStream(out));
+
+		Files.createDirectories(Paths.get("/tmp/config_pubsub_integration_test"));
+		File properties = new File("/tmp/config_pubsub_integration_test/application.properties");
+
+		try (FileOutputStream fos = new FileOutputStream(properties)) {
+			fos.write(("example.message = " + VALUE).getBytes());
+		}
+	}
+
+	@AfterClass
+	public static void tearDown() throws Exception {
+		Files.delete(Paths.get("/tmp/config_pubsub_integration_test/application.properties"));
+		Files.delete(Paths.get("/tmp/config_pubsub_integration_test"));
+		System.setOut(systemOut);
+	}
+
+	@Test
+	public void testSample() {
+
+		SpringApplicationBuilder configServer = new SpringApplicationBuilder(PubSubConfigServerApplication.class)
+			.properties("server.port=8888",
+				"spring.profiles.active=native",
+				"spring.cloud.config.server.native.searchLocations=file:/tmp/config_pubsub_integration_test/");
+		configServer.run();
+
+		Awaitility.await("config server begins watching directory")
+			.atMost(60, TimeUnit.SECONDS)
+			.until(() -> {
+				return baos.toString().contains("Monitoring for local config changes: [/tmp/config_pubsub_integration_test]");
```

Comment:
If we are monitoring for changes, do we want to test what happens when we change the file?